pkg/gofr/http/responder.go

```
@@ -53,9 +53,22 @@ func (r Responder) Respond(data any, err error) {
 		r.w.Header().Set("Content-Type", "application/json")
 	}
 
-	r.w.WriteHeader(statusCode)
+	jsonData, encodeErr := json.Marshal(resp)
+	if encodeErr != nil {
+		r.w.WriteHeader(http.StatusInternalServerError)
+
+		errorResp := response{Error: map[string]any{"message": "failed to encode response as JSON"}}
+		errorJSON, _ := json.Marshal(errorResp)
 
-	_ = json.NewEncoder(r.w).Encode(resp)
+		_, _ = r.w.Write(errorJSON)
+		_, _ = r.w.Write([]byte("\n"))
+
+		return
+	}
```

You are ignoring the error maybe you could simply hardcore things